### ISSUE TITLE
Xamarin Forms issues

### ISSUE THREAD

**Issue by Cheveyo83** [OYIon/LiveSharp#56]
Hello,
first I have to say that this is an awesome package. It improves the dev experience a lot! Thank you!

I am still new to C# and Xamarin Forms, so I'm still trying a lot of things out. I found some issues (maybe in LiveSharp, maybe in my code :-)) which I have reproduced in a simple app, to be sure.
Hints: My Xamarin App (on Windows with an Android Emulator) currently does not use any XAML so the layout is created in C# (still uncertain which I prefer). Also I use https://github.com/VincentH-Net/CSharpForMarkup for better helper functions.

1. Shell
When I use the Xamarin Forms Shell (using Tabbar) changes in the ViewModels do not get reloaded. The LiveSharp Server Status identifies changes made in the ViewModel class, however the inspector --> ViewModels Tab does not find any ViewModels at all. Also the changes are not reloaded into the Page.
Using the same page + view model as main page in the app, everything works fine.
An additional question: How is the ViewModel being recognized? Does it have to be created within the View? I try to separate the View from the ViewModel so I try to set the ViewModel dynamically from the outside during navigation.

2. Formatted Text
When using a Label with FormattedText, where a Span only contains whitespaces, then every reload via LiveSharp removes these whitespaces (trims the texts?).
E.g.:
```
	new Label () .FormattedText(
		new Span { Text = "One" },
		new Span { Text = " " },
		new Span { Text = "Two" },
		new Span { Text = "\n" },
		new Span { Text = "Three" }
	),
```	
When rebuild/redploy it renders perfectly as:
```
One Two
Three
```
However, once the Build method is run via LiveSharp, it renders as:
```
OneTwoThree
```
	

3. Styles
This may probably as well be my fault and it shoud be done in another/better way.
Currently I tried to separate the Styles into another class and just reference it in the layout class (e.g. "`new Label {} .Style(PageStyles.LabelStyle)`", where LabelStyle is a getter which always creates a new Style<Label>()).
If I make any changes in the Style class then it is recognized in LiveSharp Server Status, however, the page is not rerendered. Also if I do another change in the page afterwards and it gets rerendered, still the old styles are applied.

**Comment by ionoy** [comment]
Hello @Cheveyo83 

Thank you for the detailed feedback!

I have fixed all three issues. Can you please try the attached NuGet package to see if everything works as expected now? (You'll need to set up a local NuGet feed in the VS/Rider)

[LiveSharp.1.1.42.zip](https://github.com/OYIon/LiveSharp/files/4276332/LiveSharp.1.1.42.zip)

**Comment by Cheveyo83** [comment]
Hello @ionoy 

Thanks a lot for the quick help.

I only did some quick testing:
The ViewModel now works, when I directly create the page in the Shell. E.g.
```
Items = { new ShellContent()
{
    Content = new TestPage()
}}
```

It does not work, when the page is created lazily:
```
Items = { new ShellContent()
{
    ContentTemplate = new DataTemplate(() => new TestPage())
}}
```

As a workaround this could maybe be solved checking the DEBUG constant to use the lambda expression only in release build.


Issues 2 and 3 have also been solved!

**Comment by ionoy** [comment]
I have published NuGet package update 1.1.46 that fixes all of the issues above. Please try it when you get a chance.

Thanks!

**Comment by ionoy** [comment]
Is it possible that NRE happens in the ViewModel constructor? Can you put `Debug.WriteLine("Constructor called")` at the beginning and see if it gets called?

Thanks!

**Comment by Cheveyo83** [comment]
Yes the constructor gets called, but no the NRE does not happen in the constructor.
I only set a string property:

>         public MainPageViewModel()
>         {
>             Debug.WriteLine("Constructor called");
>             
>             Test = "test";
>             
>             Debug.WriteLine("Leaving constructor");
>         }
> 
The leaving statement is also printed. However the Build method in the Page is not called afterwards.

**Comment by ionoy** [comment]
Can you try with this package?

Thanks!

[LiveSharp.1.2.3.zip](https://github.com/OYIon/LiveSharp/files/4410021/LiveSharp.1.2.3.zip)

**Comment by Cheveyo83** [comment]
Yes, seems to work fine, now!

Thanks a lot!
